dig.go

```diff
@@ -498,9 +519,23 @@ func (c *Container) provide(ctor interface{}, opts provideOptions) error {
 		}
 		c.isVerifiedAcyclic = true
 	}
-
 	c.nodes = append(c.nodes, n)
 
+	// Record introspection info for caller if Info option is specified
+	if opts.Info != nil {
+		opts.Info.ID = (int)(n.id)
+		opts.Info.Inputs = make([]string, 0)
+		opts.Info.Outputs = make([]string, 0)
+
+		for _, res := range n.ResultList().DotResult() {
```

Comment:
If we know the number of items, we should pre-allocate the slice to avoid resizing. https://github.com/uber-go/guide/blob/master/style.md#specifying-slice-capacity

Either:

```
outputs := make([]string, 0, len(results))
for _, res := range results { outputs = append(outputs, res.Type.String()) }
```

Or,

```
outputs := make([]string, len(results))
for i, res := range results { outputs[i] = res.Type.String() }
```